### ISSUE TITLE
long function name causes "fatal error: stack corruption detected"

### ISSUE THREAD

**Issue by chriscoey** [JuliaLang/julia#29911]
on
```
Julia Version 1.1.0-DEV.605
Commit e2aba7a1bd (2018-11-02 15:19 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, sandybridge)
```
everything is fine if I run
`function arstarstarstarstarstarstarstarstarstarstarst end`
but if I try to run
`function arstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarst end`
then I get a
```
julia> function arstarstarstarstarstarstarstarstarstarstarstarstarstarstarstarstfatal error: stack corruption detected

signal (6): Aborted
in expression starting at no file:0
__libc_signal_restore_set at /build/glibc-B9XfQf/glibc-2.28/signal/../sysdeps/unix/sysv/linux/internal-signals.h:84 [inlined]
raise at /build/glibc-B9XfQf/glibc-2.28/signal/../sysdeps/unix/sysv/linux/raise.c:48
abort at /build/glibc-B9XfQf/glibc-2.28/stdlib/abort.c:79
__stack_chk_fail at /home/coey/julia/src/codegen.cpp:123
fl_accum_julia_symbol at /home/coey/julia/src/flisp/julia_extensions.c:325
apply_cl at /home/coey/julia/src/flisp/flisp.c:1176
do_trycatch at /home/coey/julia/src/flisp/flisp.c:898
apply_cl at /home/coey/julia/src/flisp/flisp.c:1806
do_trycatch at /home/coey/julia/src/flisp/flisp.c:898
apply_cl at /home/coey/julia/src/flisp/flisp.c:1806
do_trycatch at /home/coey/julia/src/flisp/flisp.c:898
apply_cl at /home/coey/julia/src/flisp/flisp.c:1806
do_trycatch at /home/coey/julia/src/flisp/flisp.c:898
apply_cl at /home/coey/julia/src/flisp/flisp.c:1806
_applyn at /home/coey/julia/src/flisp/flisp.c:682
fl_applyn at /home/coey/julia/src/flisp/flisp.c:722
jl_parse_input_line at /home/coey/julia/src/ast.c:774
#717 at ./client.jl:156 [inlined]
with_logstate at ./logging.jl:395
with_logger at ./logging.jl:491 [inlined]
#parse_input_line#716 at ./client.jl:155 [inlined]
#parse_input_line at ./none:0 [inlined]
return_callback at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:676
jfptr_return_callback_12991 at /home/coey/julia/usr/lib/julia/sys.so (unknown line)
jl_apply_generic at /home/coey/julia/src/gf.c:2209
on_enter at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:1870
on_enter at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:201
#109 at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:1993
jl_apply_generic at /home/coey/julia/src/gf.c:2209
jl_apply at /home/coey/julia/src/julia.h:1571 [inlined]
jl_f__apply at /home/coey/julia/src/builtins.c:556
jl_f__apply_latest at /home/coey/julia/src/builtins.c:594
#invokelatest#1 at ./essentials.jl:697 [inlined]
invokelatest at ./essentials.jl:696 [inlined]
#27 at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:1319
jl_apply_generic at /home/coey/julia/src/gf.c:2209
prompt! at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:2360
run_interface at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:2263
jl_apply_generic at /home/coey/julia/src/gf.c:2209
run_frontend at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:1035
run_repl at /home/coey/julia/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:192
jl_apply_generic at /home/coey/julia/src/gf.c:2209
#723 at ./logging.jl:309
jfptr_#723_5954 at /home/coey/julia/usr/lib/julia/sys.so (unknown line)
jl_apply_generic at /home/coey/julia/src/gf.c:2209
jl_apply at /home/coey/julia/src/julia.h:1571 [inlined]
jl_f__apply at /home/coey/julia/src/builtins.c:556
jl_f__apply_latest at /home/coey/julia/src/builtins.c:594
#invokelatest#1 at ./essentials.jl:697 [inlined]
invokelatest at ./essentials.jl:696 [inlined]
macro expansion at ./logging.jl:306 [inlined]
run_main_repl at ./client.jl:334
exec_options at ./client.jl:246
_start at ./client.jl:425
jl_apply_generic at /home/coey/julia/src/gf.c:2209
jl_apply at /home/coey/julia/ui/../src/julia.h:1571 [inlined]
true_main at /home/coey/julia/ui/repl.c:96
main at /home/coey/julia/ui/repl.c:217
__libc_start_main at /build/glibc-B9XfQf/glibc-2.28/csu/../csu/libc-start.c:308
_start at /home/coey/julia/julia (unknown line)
Allocations: 5309928 (Pool: 5308846; Big: 1082); GC: 10
Aborted (core dumped)
```

**Comment by chriscoey** [comment]
woops I guess this is due to https://github.com/JuliaLang/julia/issues/29903#event-1943435906
will try that and close if it fixes

**Comment by chriscoey** [comment]
confirmed that the fix in #29903 worked
